### PR TITLE
[el-date-picker]: Fix Accessibility: DateTime picker must be keyboard…

### PR DESCRIPTION
**Pull request by abhipanda**
… and VoiceOver reachable #13191

Please make sure these boxes are checked before submitting your PR, thank you!

* [x] Make sure you follow Element's contributing guide ([中文](https://github.com/ElemeFE/element/blob/master/.github/CONTRIBUTING.zh-CN.md) | [English](https://github.com/ElemeFE/element/blob/master/.github/CONTRIBUTING.en-US.md) | [Español](https://github.com/ElemeFE/element/blob/master/.github/CONTRIBUTING.es.md)).
* [x] Make sure you are merging your commits to `dev` branch.
* [x] Add some descriptions and refer relative issues for you PR.

### CODE REVIEW

File: packages/date-picker/src/panel/date.vue [hunk 1]
```diff
@@ -52,35 +52,41 @@
...
               type="button"
               @click="prevMonth"
+              @keyup.enter="prevMonth"
               v-show="currentView === 'date'"
               :aria-label="t(`el.datepicker.prevMonth`)"
               class="el-picker-panel__icon-btn el-date-picker__prev-btn el-icon-arrow-left">
             </button>
-            <span
+            <button
               @click="showYearPicker"
```
Comment (wacky6) [inline packages/date-picker/src/panel/date.vue]: use `click.stop`, do not add event param to {next, prev}{Year, Month} and show{Year, Month}Picker methods.

File: packages/date-picker/src/panel/date.vue [hunk 2]
```diff
@@ -125,6 +131,7 @@
           size="mini"
           type="text"
           class="el-picker-panel__link-btn"
+          aria-haspopup="true" aria-expanded="false"
```
Comment (wacky6) [inline packages/date-picker/src/panel/date.vue]: One line per attribute.

File: packages/date-picker/src/panel/date.vue [hunk 3]
```diff
@@ -407,6 +421,7 @@
             event.stopPropagation();
             event.preventDefault();
           }
+          // this.$refs.dateInput.focus();
```
Comment (wacky6) [inline packages/date-picker/src/panel/date.vue]: Remove commented code if it is unused.

File: packages/date-picker/src/picker.vue [hunk 4]
```diff
@@ -1,5 +1,6 @@
 <template>
-  <el-input
+  <el-input aria-haspopup="true"
```
Comment (wacky6) [inline packages/date-picker/src/picker.vue]: New line for this attribute; Fix indentation

```
<el-input
  aria-haspopup="true"
```

File: packages/date-picker/src/picker.vue [hunk 5]
```diff
@@ -48,6 +49,8 @@
     v-else>
     <i :class="['el-input__icon', 'el-range__icon', triggerClass]"></i>
     <input
+            aria-haspopup="true"
```
Comment (wacky6) [inline packages/date-picker/src/picker.vue]: Fix indentation;

File: packages/date-picker/src/picker.vue [hunk 6]
```diff
@@ -128,6 +133,9 @@ const HAVE_TRIGGER_TYPES = [
   'datetimerange',
   'dates'
 ];
+/**
```
Comment (wacky6) [inline packages/date-picker/src/picker.vue]: Remove this comment.

File: packages/date-picker/src/picker.vue [hunk 7]
```diff
@@ -136,6 +144,9 @@ const DATE_PARSER = function(text, format) {
   if (format === 'timestamp') return new Date(Number(text));
   return parseDate(text, format);
 };
+/**
```
Comment (wacky6) [inline packages/date-picker/src/picker.vue]: Remove this comment

File: packages/date-picker/src/picker.vue [hunk 8]
```diff
@@ -412,8 +423,8 @@ export default {
         this.emitChange(this.value);
         this.userInput = null;
         this.dispatch('ElFormItem', 'el.form.blur');
-        this.$emit('blur', this);
-        this.blur();
+        // this.$emit('blur', this);
```
Comment (wacky6) [inline packages/date-picker/src/picker.vue]: Why is this code commented?

File: packages/date-picker/src/picker.vue [hunk 9]
```diff
@@ -742,7 +753,7 @@ export default {
         if (!this.ranged) {
           this.handleChange();
           this.pickerVisible = this.picker.visible = false;
-          this.blur();
+          // this.blur();
```
Comment (wacky6) [inline packages/date-picker/src/picker.vue]: Why is this commented

File: packages/theme-chalk/src/date-picker/date-picker.scss [hunk 10]
```diff
@@ -64,6 +64,8 @@
     line-height: 22px;
     text-align: center;
     cursor: pointer;
+    background: none;
```
Comment (wacky6) [inline packages/theme-chalk/src/date-picker/date-picker.scss]: Why is css change necessary

Comment (abhipanda) [reply packages/theme-chalk/src/date-picker/date-picker.scss]: Original span converted to button - which is why need to remove background

File: packages/date-picker/src/picker.vue (hunk 9 above)
Comment (abhipanda) [reply packages/date-picker/src/picker.vue]: This allows the focus to remain on the triggering input text box and next tabbing takes the user to next in line tab index

File: packages/date-picker/src/picker.vue (hunk 8 above)
Comment (abhipanda) [reply packages/date-picker/src/picker.vue]: This allows the focus to remain on the triggering input text box and next tabbing takes the user to next in line tab index

**Pull request closed without merging**